### DISCUSSION TITLE
Documentation on network restrictions?

### DISCUSSION THREAD

**Discussion by JeremyRand** [cirruslabs/cirrus-ci-docs#955] [Q&A] [answered]
Is there any documentation on what types of network activity are and are not allowed?  All I can find is the "Repository is blocked" FAQ, which mentions cryptocurrency mining, piracy, and ad fraud.

I briefly tried to set up integration tests for a cryptocurrency repo that connect to public ElectrumX servers (obviously no mining), and the TCP connections all seem to be failing, which makes me wonder whether I've inadvertently triggered some kind of abuse detection heuristic that blocks network activity.

Log is at https://cirrus-ci.com/task/6236556557025280 (see the `Run electrum` section; the log includes the hostnames/IP's it's trying to connect to if that helps you compare with firewall rules or something).

**Comment by fkorotkov** [comment, marked as answer]
There was a huge increase in crypto mining attempts on all services that provide compute for free in 2021 including Cirrus CI. Multiple companies just drastically restricted free usage:

* [GitLab](https://about.gitlab.com/blog/2021/05/17/prevent-crypto-mining-abuse/) started requiring a credit card and implemented monthly CI quotas.
* [Docker](https://www.docker.com/blog/changes-to-docker-hub-autobuilds) completely disabled Autobuilds for free users. 
* GitHub implemented some restrictions that they [announced publicly](https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/) and a few that they haven't announced.

We on the other hand decided not to restrict any usage of Cirrus CI and implemented multiple different mechanism to pro-actively fight with crypto mining on the clusters that we provide the free service on. One of them analyzes and blocks some network activities. We can't go much into the exact details of now our anti-mining safeguards work to not spoil them and make life of miners easier.

As for your use case, it seems it won't be possible for the mechanism to distinguish between you trying to run integration tests and a potential miner trying to connect to ElectrumX in order to mine. The only workaround I see is for you to use some [instance that is managed by you](https://cirrus-ci.org/guide/writing-tasks/#execution-environment). Cirrus CI is not running the anti-mining safeguards on the connected [compute services](https://cirrus-ci.org/guide/supported-computing-services/).

**Reply by JeremyRand** [reply]
I see, thanks for the explanation.  This is regrettable but understandable.
